src/cmd/ksh93/sh/deparse.c

```diff
@@ -305,6 +306,7 @@ static_fn void p_tree(const Shnode_t *t, int tflags) {
                 }
                 p_arg(&(t->lst.lstlef->arg), ' ', 0);
                 if (t->tre.tretyp & TBINARY) {
+                    assert(cp != NULL);
```

Comment:
Why ?

Reply:
Because the next statement, `sfputr(outfile, cp, ' ');`, uses that var and expects it to be non-NULL. However, the way the code is structured there is theoretically a path that reaches here wherein `cp` is not set to a non-NULL value. The assert tells the compiler we expect it to be non-NULL and silences the warning.